aten/src/ATen/native/Convolution.cpp

```diff
@@ -207,6 +218,43 @@ static void check_input_shape_forward(const at::Tensor& input,
              "Given weight of size ", weight.sizes(),
              ", expected bias to be 1-dimensional with ", weight.size(0), " elements",
              ", but got bias of size ", bias.sizes(), " instead");
+
+    for (int i = 2; i < k; ++i) {
+      input_shape.push_back(input.size(i) + 2 * padding[i-2]);
+      kernel_shape.push_back(weight.size(i));
+      if (input_shape.back() < kernel_shape.back()) {
+        kernel_size_correct = false;
+      }
+
+      // TODO: Should output padding be considered while computing output_shape?
+      output_shape.push_back(div_rtn<int64_t>(input_shape.back() - kernel_shape.back(), stride[i-2]) + 1);
```

Comment:
I believe dilation should also be taken into account here

Reply:
@fmassa do we claim to support negative stride for convolutional layers?

Reply:
I agree with @mrshenli 's previous comment of not allowing negative stride which removes the need of output_shape only if we consider dilation for updating new kernel size and use it for input shape mismatch

Reply:
Sounds good to me